modin/engines/ray/pandas_on_ray/frame/partition.py

```diff
@@ -45,33 +47,22 @@ def apply(self, func, **kwargs):
             A RayRemotePartition object.
         """
         oid = self.oid
-        self.call_queue.append((func, kwargs))
-
-        def call_queue_closure(oid_obj, call_queues):
-
-            for func, kwargs in call_queues:
-                if isinstance(func, ray.ObjectID):
-                    func = ray.get(func)
-                if isinstance(kwargs, ray.ObjectID):
-                    kwargs = ray.get(kwargs)
-
-                oid_obj = func(oid_obj, **kwargs)
-
-            return oid_obj
-
-        oid = deploy_ray_func.remote(
-            call_queue_closure, oid, kwargs={"call_queues": self.call_queue}
+        call_queue = self.call_queue + [(func, kwargs)]
+        new_obj, result, length, width = deploy_ray_func.remote(call_queue, oid)
+        if len(self.call_queue) > 0:
+            self.oid = new_obj
+            self.call_queue = []
+        return PandasOnRayFramePartition(
+            result, PandasOnRayFramePartition(length), PandasOnRayFramePartition(width)
```

Comment:
Why are we setting the length and width to a `PandasOnRayFramePartition` object with an object id of `length` and `width`, respectively?

Reply:
It is kind of confusing, but basically this is to be compatible with other code.

I will add a comment.